pkg/controller/controller.go

```diff
@@ -277,13 +288,23 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 		req.ControllerCreateSecrets = credentials
 		secret.Name, secret.Namespace, _ = getSecretAndNamespaceFromParameters(options.Parameters)
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), p.timeout)
-	defer cancel()
 
-	rep, err := p.csiClient.CreateVolume(ctx, &req)
+	opts := wait.Backoff{Factor: backoffFactor, Steps: backoffSteps}
```

Comment:
Dumb question, since I am not familiar with the provisioner code, does this path happen on a separate go routine? Or does it block the provisioner from processing other PVCs?

If it blocks other provisioners, we should change the model such that we have a separate goroutine per volume (maybe using https://github.com/kubernetes/kubernetes/blob/master/pkg/util/goroutinemap/goroutinemap.go).

Reply:
Current implementation is a single go routine. I will look into converting it multiple go routines.

Reply:
I checked the provisioner controller code and it seem each Provision action would be run in a separate go routine. So it seems to be safe to block one specific one while it waits for a driver response.